gen-PKGBUILD.py

```diff
@@ -7,28 +7,287 @@
+		),
+
+		'amdgpu-pro-vulkan': Package(
+			desc = "The AMDGPU Pro Vulkan driver",
+			provides = ['vulkan-driver'],
+			extra_commands = [
+				'mkdir -p "${pkgdir}"/usr/share/vulkan/icd.d/',
+				'mv "${pkgdir}"/etc/vulkan/icd.d/amd_icd64.json "${pkgdir}"/usr/share/vulkan/icd.d/',
+				'rm -rf "${pkgdir}"/etc/vulkan/'
+			]
+		),
+
+		'amdgpu-pro-vdpau': Package(
+			desc = "The AMDGPU Pro VDPAU driver",
+			extra_commands = [
+				'mkdir -p "${pkgdir}"/usr/lib/',
+				'ln -s /opt/amdgpu-pro/lib/x86_64-linux-gnu/vdpau/libvdpau_amdgpu.so.1.0.0 "${pkgdir}"/usr/lib/libvdpau_amdgpu.so.1.0.0',
+				'ln -s /opt/amdgpu-pro/lib/x86_64-linux-gnu/vdpau/libvdpau_amdgpu.so.1.0.0 "${pkgdir}"/usr/lib/libvdpau_amdgpu.so.1',
+				'ln -s /opt/amdgpu-pro/lib/x86_64-linux-gnu/vdpau/libvdpau_amdgpu.so.1.0.0 "${pkgdir}"/usr/lib/libvdpau_amdgpu.so',
+			]
+		),
+
+		'lib32-amdgpu-pro': Package(
+			desk = "The AMDGPU Pro driver package (32bit libraries)",
+			extra_commands = [
+				'mkdir -p "${pkgdir}"/usr/lib32/',
+				'mv "${pkgdir}"/usr/lib/i386-linux-gnu/dri "${pkgdir}"/usr/lib32/',
+
+				'rm -rf "${pkgdir}"/etc',
+				'mkdir -p "${pkgdir}/etc/ld.so.conf.d/"',
+				'echo "/opt/amdgpu-pro/lib/i386-linux-gnu/" > "${pkgdir}"/etc/ld.so.conf.d/lib32-amdgpu-pro.conf'
+			]
+		),
+
+		'lib32-amdgpu-pro-opencl': Package(
+			desc = "The AMDGPU Pro OpenCL implementation",
+			provides  = ['lib32-opencl-driver']
+		),
+
+		'lib32-amdgpu-pro-vulkan': Package(
+			desc = "The AMDGPU Pro Vulkan driver (32bit libraries)",
+			provides = ['lib32-vulkan-driver'],
+			extra_commands = [
+				'mkdir -p "${pkgdir}"/usr/share/vulkan/icd.d/',
+				'mv "${pkgdir}"/etc/vulkan/icd.d/amd_icd32.json "${pkgdir}"/usr/share/vulkan/icd.d/',
+				'rm -rf "${pkgdir}"/etc/vulkan/'
+			]
+		),
+
+		'lib32-amdgpu-pro-libdrm': Package(
+			desc = "The AMDGPU Pro userspace interface to kernel DRM services (32bit libraries)",
+			conflicts = ['lib32-libdrm'],
+			provides = ['lib32-libdrm'],
+		),
+
+		'lib32-amdgpu-pro-libgl': Package(
+			desc = "The AMDGPU Pro libgl library symlinks (32bit libraries)",
+			conflicts = ['lib32-libgl'],
+			provides  = ['lib32-libgl'],
+			depends   = ['lib32-amdgpu-pro'],
+		),
+
+		'lib32-amdgpu-pro-vdpau': Package(
+			desc = "The AMDGPU Pro VDPAU driver (32bit libraries)",
+			extra_commands = [
+				'mkdir -p "${pkgdir}"/usr/lib32/',
+				'ln -s /opt/amdgpu-pro/lib/i386-linux-gnu/vdpau/libvdpau_amdgpu.so.1.0.0 "${pkgdir}"/usr/lib32/libvdpau_amdgpu.so.1.0.0',
+				'ln -s /opt/amdgpu-pro/lib/i368-linux-gnu/vdpau/libvdpau_amdgpu.so.1.0.0 "${pkgdir}"/usr/lib32/libvdpau_amdgpu.so.1',
+				'ln -s /opt/amdgpu-pro/lib/i368-linux-gnu/vdpau/libvdpau_amdgpu.so.1.0.0 "${pkgdir}"/usr/lib32/libvdpau_amdgpu.so',
+			]
+		),
+
+		'xf86-video-amdgpu-pro': Package(
+			desc = "The AMDGPU Pro X.org video driver",
+			conflicts = ['xf86-video-amdgpu', 'xorg-server<1.18.0', 'xorg-server>=1.19.0' 'X-ABI-VIDEODRV_VERSION<20', 'X-ABI-VIDEODRV_VERSION>=21'],
```

Comment:
@brainpower I just noticed this typo here (missing `,`).  It's not an urgent problem as it looks like it will only break the upper bound of xorg-server and the lower bound of the XBI version.